pkg/util/kubernetes/apiserver/apiserver.go

```diff
@@ -98,14 +101,41 @@ func getKubeClient(timeout time.Duration) (kubernetes.Interface, error) {
 			return nil, err
 		}
 	}
-	clientConfig.Timeout = timeout
 
 	if config.Datadog.GetBool("kubernetes_apiserver_use_protobuf") {
 		clientConfig.ContentType = "application/vnd.kubernetes.protobuf"
 	}
+	return clientConfig, nil
+}
+
+func getKubeClient(timeout time.Duration) (kubernetes.Interface, error) {
+	clientConfig, err := getClientConfig()
+	if err != nil {
+		return nil, err
+	}
+	clientConfig.Timeout = timeout
 	return kubernetes.NewForConfig(clientConfig)
 }
 
+func getWPAClient(timeout time.Duration) (wpa_client.Interface, error) {
+	clientConfig, err := getClientConfig()
+	if err != nil {
+		return nil, err
+	}
+	clientConfig.Timeout = timeout
+	return wpa_client.NewForConfig(clientConfig)
+}
+
+func getWPAInformerFactory() (wpa_informers.SharedInformerFactory, error) {
+	resyncPeriodSeconds := time.Duration(config.Datadog.GetInt64("kubernetes_informers_resync_period"))
```

Comment:
can we set a default value?

Reply:
Yes it defaults at 300s https://github.com/DataDog/datadog-agent/blob/master/pkg/config/config.go#L423